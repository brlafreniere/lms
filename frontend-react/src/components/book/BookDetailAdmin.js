import React from "react";
import {useParams} from "react-router-dom";

import Book from "../../modules/book";

export default function BookDetailAdmin(props) {
    let {id} = useParams();
    return (
        <BookDetailAdminComponent id={id} />
    )
}

export class BookDetailAdminComponent extends React.Component {
    constructor(props) {
        super(props)
        this.state = {
            book: {},
            book_inventories: []
        }
    }

    componentDidMount() {
        Book.fetch(this.props.id).then(book => {
            this.setState({book: book})
        }).catch(console.log);
        Book.fetchInventories(this.props.id).then(book_inventories => {
            this.setState({book_inventories: book_inventories})
        }).catch(console.log)
    }

    render () {
        return (
            <div>
                <h2>{this.state.book.title}</h2>
                <div>
                    {this.state.book.synopsis}
                </div>
                <div className="mt-5">
                    <h3>Inventory</h3>
                    <table className="table">
                        <thead>
                            <tr>
                                <th>Branch</th>
                                <th>Copies</th>
                            </tr>
                        </thead>
                        <tbody>
                            {this.state.book_inventories.map(bi => (
                                <tr key={bi.id}>
                                    <td>{bi.branch.name}</td>
                                    <td>{bi.copies}</td>
                                </tr>
                            ))}
                        </tbody>
                    </table>
                </div>
            </div>
        )
    }
}