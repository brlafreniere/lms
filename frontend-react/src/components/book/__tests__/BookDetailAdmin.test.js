import React from "react";
import renderer from "react-test-renderer";
import BookDetailAdmin from "../BookDetailAdmin";

import {MemoryRouter} from "react-router-dom";
import Book from "../../../modules/book";

test("<BookDetailAdmin /> snapshot test", () => {
    const book = {
        id: 4,
        title: "foo title bar",
        synopsis: "lorem ipsum dolar sit amet"
    }

    Book.fetch = jest.fn()
    Book.fetch.mockResolvedValue(book)

    const component = renderer.create(
        <MemoryRouter>
            <BookDetailAdmin id={4} />
        </MemoryRouter>
    )

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})