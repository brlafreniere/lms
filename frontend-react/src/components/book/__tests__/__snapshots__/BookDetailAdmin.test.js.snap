// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BookDetailAdmin /> snapshot test 1`] = `
<div>
  <h2 />
  <div />
  <div
    className="mt-5"
  >
    <h3>
      Inventory
    </h3>
    <table
      className="table"
    >
      <thead>
        <tr>
          <th>
            Branch
          </th>
          <th>
            Copies
          </th>
        </tr>
      </thead>
      <tbody />
    </table>
  </div>
</div>
`;
